src/fabric.c

```diff
@@ -977,7 +977,9 @@ void DEFAULT_SYMVER_PRE(fi_freeinfo)(struct fi_info *info)
 		}
 		if (info->fabric_attr) {
 			free(info->fabric_attr->name);
-			free(info->fabric_attr->prov_name);
+			if (info->fabric_attr->prov_name == "sockets" ||
+                            info->fabric_attr->prov_name == NULL)
+				free(info->fabric_attr->prov_name);
```

Comment:
See the comment on the previous patch about using strdup.  That's likely leading to this crash.

We shouldn't have provider specific checks in this common path.  This likely fixes the crash by instead leaking memory.